src/engine/shared/network_server.cpp

```diff
@@ -191,7 +192,7 @@ int CNetServer::Recv(CNetChunk *pChunk, TOKEN *pResponseToken)
 								char aBuf[128];
 								str_format(aBuf, sizeof(aBuf), "Only %d players with the same IP are allowed", m_MaxClientsPerIP);
 								SendControlMsg(&Addr, m_RecvUnpacker.m_Data.m_ResponseToken, 0, NET_CTRLMSG_CLOSE, aBuf, str_length(aBuf) + 1);
-								return 0;
+								goto loopbegin;
```

Comment:
should be a continue i think?

Reply:
Can't be a continue, the continue needs to happen in the outer loop.

Reply:
Or we can introduce an extra boolean flag, set it and hope the compiler is smart enough to optimize it out. You know what I think I'll do that, adhering to some arbitrary set of rules set to keep under-qualified people writing code for mission critical hardware from making mistakes is more important then writing short and efficient code.